include/eve/algo/container/soa_vector2.hpp

```diff
@@ -0,0 +1,371 @@
+    //! @}
+    //==============================================================================================
+
+    //==============================================================================================
+    //! @name Capacity
+    //! @{
+    //==============================================================================================
+    //! Returns the number of elements in the container
+    EVE_FORCEINLINE std::size_t size()  const noexcept { return size_; }
+
+    //! Returns the number of elements that the container has currently allocated space for
+    EVE_FORCEINLINE std::size_t capacity()  const noexcept { return storage.capacity_; }
+
+    //! Requests the removal of unused capacity.
+    EVE_FORCEINLINE void shrink_to_fit() { grow(size(), size()); }
+
+    //! Checks if the container has no elements
+    EVE_FORCEINLINE bool empty() const noexcept { return !size(); }
+
+    //! @brief Increase the capacity of the vector to a value that's greater or equal to `n`.
+    //! If `n` is greater than the current capacity(), new storage is allocated and all iterators,
+    //! including the past-the-end iterator, and all references to the elements are invalidated.
+    //! Otherwise, the method does nothing.
+    //! @warning reserve() does not change the size of the vector.
+    //! @param n New capacity of the `eve::soa_vector`
+    void reserve(std::size_t n)
+    {
+      if(n > capacity())
+      {
+        // Grow capacity to n
+        auto const sz = size_;
+        grow(n, size());
+        size_ = sz;
+     }
+    }
+
+    //==============================================================================================
+    //! @}
+    //==============================================================================================
+
+    //==============================================================================================
+    //! @name Modifiers
+    //! @{
+    //==============================================================================================
+    //! @brief Clear the contents of the container.
+    void clear() { size_ = 0; }
+
+    //! @brief Removes an element from the container.
+    //! Has the same invalidation semantics as std::vector.
+    //! end() iterator is not a valid pos.
+    iterator erase(const_iterator pos) { return erase(pos, pos+1); }
+
+    //! @brief Removes the elements in the range [first, last)
+    //! Empty range is OK, does nothing
+    iterator erase(const_iterator f, const_iterator l)
+    {
+      EVE_ASSERT(!empty(), "soa_vector::erase() was called on empty container");
+
+      std::ptrdiff_t distance_f = f - cbegin();
+      std::ptrdiff_t distance_l = l - cbegin();
+      std::ptrdiff_t sz         = distance_l - distance_f;
+
+      eve::algo::copy_backward( algo::as_range(l, cend())
+                              , algo::as_range(begin() + distance_l - sz, end() - sz)
```

Comment:
begin_aligned. It's saves cycles, like a lot.

Reply:
again, begin_aligned() + distance_l - sz has no chance being aligned and will assert at runtime that the address is not aligned

Reply:
You are correct, I am sorry.